sdk/src/androidTest/java/com/karhoo/sdk/api/CheckCoverageIntegrationTest.kt

```diff
@@ -0,0 +1,228 @@
+package com.karhoo.sdk.api
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.tomakehurst.wiremock.junit.WireMockRule
+import com.karhoo.sdk.api.model.Coverage
+import com.karhoo.sdk.api.network.request.CoverageRequest
+import com.karhoo.sdk.api.network.response.Resource
+import com.karhoo.sdk.api.testrunner.SDKTestConfig
+import com.karhoo.sdk.api.util.ServerRobot.Companion.COVERAGE_OK
+import com.karhoo.sdk.api.util.ServerRobot.Companion.EMPTY
+import com.karhoo.sdk.api.util.ServerRobot.Companion.GENERAL_ERROR
+import com.karhoo.sdk.api.util.ServerRobot.Companion.INVALID_JSON
+import com.karhoo.sdk.api.util.ServerRobot.Companion.LAT
+import com.karhoo.sdk.api.util.ServerRobot.Companion.LONG
+import com.karhoo.sdk.api.util.ServerRobot.Companion.NO_BODY
+import com.karhoo.sdk.api.util.serverRobot
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.net.HttpURLConnection.HTTP_OK
+import java.net.HttpURLConnection.HTTP_UNAUTHORIZED
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@RunWith(AndroidJUnit4::class)
+class CheckCoverageIntegrationTest {
+
+    @get:Rule
+    var wireMockRule = WireMockRule(SDKTestConfig.wireMockOptions)
+
+    private val latch = CountDownLatch(1)
+
+    @Before
+    fun setup() {
+        serverRobot {
+            successfulToken()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        wireMockRule.resetAll()
+    }
+
+    /**
+     * Given:   Check coverage is requested
+     * When:    Successful response has been returned
+     * Then:    The response payload should be valid
+     **/
+    @Test
+    fun checkCoverageSuccess() {
+        serverRobot {
+            getCheckCoverageResponse(code = HTTP_OK, response = COVERAGE_OK)
+        }
+        var result: Coverage? = null
+
+        KarhooApi.quotesService.checkCoverage(CoverageRequest(LAT, LONG, "")).execute {
```

Comment:
Think it might be worth just creating one CoverageRequest and reusing as the values don't change.

Reply:
Done. Made code much cleaner 👍